bin/verify_aws_cli_setup

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash -eu
+
+echo "--- :aws: Ensuring AWS CLI is installed and configured..."
+
+echo "~~~ Checking if AWS CLI is installed..."
+if ! command -v aws &> /dev/null; then
+  echo "[!] AWS CLI not found."
+  exit 1
+else
+  echo "AWS CLI found at $(which aws)."
+  aws --version
+fi
+
+echo "~~~ Checking if AWS credentials are configured in env vars or config files..."
+
+if [ -n "${AWS_ACCESS_KEY_ID:-}" ]; then
+    echo "AWS_ACCESS_KEY_ID found in environment variables."
+else
+    echo "No AWS_ACCESS_KEY_ID in environment variables."
+fi
+
+if [ -n "${AWS_SECRET_ACCESS_KEY:-}" ]; then
+    echo "AWS_SECRET_ACCESS_KEY found in environment variables."
+else
+    echo "No AWS_SECRET_ACCESS_KEY in environment variables."
+fi
+
+AWS_CONFIG_PATH="$HOME/.aws/config"
+if [ -f "$AWS_CONFIG_PATH" ]; then
+    echo "AWS config file exists at $AWS_CONFIG_PATH."
+    echo "Printing config via aws configure list:"
+    aws configure list
+else
+    echo "No AWS config file found at $AWS_CONFIG_PATH."
+fi
+
+AWS_CREDENTIALS_PATH="$HOME/.aws/credentials"
+if [ -f "$AWS_CREDENTIALS_PATH" ]; then
+    echo "AWS credentials file exists at $AWS_CREDENTIALS_PATH."
+    echo "Printing profiles via aws configure list:"
+    aws configure list-profiles
+else
+    echo "No AWS credentials file found at $AWS_CREDENTIALS_PATH."
+fi
+
+echo "~~~ Checking AWS credentials via CLI call..."
+if ! aws sts get-caller-identity &> /dev/null; then
```

Comment:
My first reaction to this was wondering what would happen if the EC2 instance [has the proper role to access the S3 bucket](https://github.com/Automattic/buildkite-ci/blob/trunk/src/aws/access/policies/can_use_ci_cache.tf) we need it to access (`s3://a8c-ci-cache`), but did not have the role/policy to perform that STS call.

But [looking at the docs](https://docs.aws.amazon.com/cli/latest/reference/sts/get-caller-identity.html) it appears  that there's no permission needed to call this one 👍  So we should be good on that front at least.